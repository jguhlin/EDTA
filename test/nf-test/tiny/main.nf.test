nextflow_pipeline {

    name "Test with a tiny genome"
    script "main.nf"

    test("tiny genome") {

        when {
            params {
                genomes = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta"
                outdir  = "$outputDir"
            }
        }

        then {
            def stable_path = getAllFilesFromDir(params.outdir, false, ['pipeline_info/*.{html,json,txt,yml}'], null, ['**'])

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    [
                        'successful tasks': workflow.trace.succeeded().size(),
                        'versions': removeNextflowVersion("$outputDir/pipeline_info/software_versions.yml"),
                        'stable paths': stable_path
                    ]
                ).match() }
            )
        }

    }

}
