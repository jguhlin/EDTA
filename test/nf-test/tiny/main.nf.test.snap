{
    "tiny genome": {
        "content": [
            {
                "successful tasks": 13,
                "versions": {
                    "ANNOSINE": {
                        "annosine": "2.0.7"
                    },
                    "CAT_CAT": {
                        "pigz": "2.3.4"
                    },
                    "HELITRONSCANNER_DRAW": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_DRAW_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_HEAD": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_HEAD_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "LTRFINDER": {
                        "LTR_FINDER_parallel": "v1.1",
                        "ltr_finder": "v1.07"
                    },
                    "LTRHARVEST": {
                        "LTR_HARVEST_parallel": "v1.1",
                        "genometools": "1.6.5"
                    },
                    "LTRRETRIEVER_LTRRETRIEVER": {
                        "LTR_retriever": "v2.9.9"
                    },
                    "TIRLEARNER": {
                        "TIR-Learner": "v3.0.2 by Tianyu (Sky) Lu (tlu83@wisc.edu) published under GPLv3"
                    },
                    "Workflow": {
                        "jguhlin/EDTA": "v0.1.0dev"
                    }
                },
                "stable paths": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T22:35:05.489004"
    }
}